### PR TITLE
binlog bug fixes – stripping checksums and applying `Query` flags

### PR DESCRIPTION
**Pull request by fulghum**
This PR fixes a couple of issues from binlog testing:

### `Query` flags
Now that we have visibility to the additional flags in the `Query` binlog event (https://github.com/dolthub/vitess/pull/220), this PR adds handling for those flags. 

### Checksum support
To receive messages from the primary when `binlog_checksums` are enabled, the replica now sends a handshake signal to the primary and strips off any received checksum on binlog events. This lets us remove the requirement to disable checksums on the primary in order to use Dolt's binlog replication. 

### Password auth for tests
This PR also includes a change in the test code for the replica connect to the primary with a password, to ensure we're testing password auth.

### CODE REVIEW

File: go/libraries/doltcore/sqle/binlogreplication/binlog_replica_applier.go [hunk 1]
```diff
@@ -337,6 +362,38 @@ func (a *binlogReplicaApplier) processBinlogEvent(ctx *sql.Context, engine *gms.
...
+			ctx.GetLogger().Tracef("Setting foreign_key_checks=0")
+			ctx.SetSessionVariable(ctx, "foreign_key_checks", 0)
+		} else {
+			ctx.GetLogger().Tracef("Setting foreign_key_checks=1")
+			ctx.SetSessionVariable(ctx, "foreign_key_checks", 1)
+		}
+
+		if query.Options&mysql.QFlagOptionRelaxedUniqueChecks > 0 {
+			ctx.GetLogger().Tracef("Setting unique_checks=0")
+			ctx.SetSessionVariable(ctx, "unique_checks", 0)
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/binlogreplication/binlog_replica_applier.go:391]: This is unimplemented in dolt, might want to note that

File: go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go [hunk 2]
```diff
@@ -486,8 +490,8 @@ func stopDoltSqlServer(t *testing.T) {
 func startReplication(_ *testing.T, port int) {
 	replicaDatabase.MustExec("SET @@GLOBAL.server_id=123;")
 	replicaDatabase.MustExec(
-		fmt.Sprintf("change replication source to SOURCE_HOST='localhost', SOURCE_USER='root', "+
-			"SOURCE_PASSWORD='', SOURCE_PORT=%v;", port))
+		fmt.Sprintf("change replication source to SOURCE_HOST='localhost', SOURCE_USER='replicator', "+
+			"SOURCE_PASSWORD='password', SOURCE_PORT=%v;", port))
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go:494]: Make this slightly less generic?

File: go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go [hunk 3]
```diff
@@ -623,9 +626,11 @@ func startMySqlServer(dir string) (int, *os.Process, error) {
 	primaryDatabase = sqlx.MustOpen("mysql", dsn)
 
 	os.Chdir(originalCwd)
-
 	fmt.Printf("MySQL server started on port %v \n", mySqlPort)
 
+	primaryDatabase.MustExec("CREATE USER 'replicator'@'%' IDENTIFIED BY 'password';")
+	primaryDatabase.MustExec("GRANT ALL PRIVILEGES ON *.* TO 'replicator'@'%';")
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go:632]: Is this grant too broad?

**Review by zachmu — APPROVED** [review]
LGTM, just a couple comments / concerns

File: go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go (hunk 3 above)
Comment (fulghum) [reply go/libraries/doltcore/sqle/binlogreplication/binlog_replication_test.go:632]: Yup – switched to `GRANT REPLICATION SLAVE`

**Pull request merged**